db.js

```diff
@@ -426,6 +429,53 @@ exports.init = function (sbot, config) {
     )
   }
 
+  function addButtwoo(buffer, cb) {
+    const guard = guardAgainstDuplicateLogs('addButtwoo()')
+    if (guard) return cb(guard)
+
+    onceWhen(
+      stateFeedsReady,
+      (ready) => ready === true,
+      () => {
+        const data = butt2.extractData(buffer)
+        if (data.length !== 2)
+          return cb('incorrect buttwoo format, addButtwoo() failed')
+        if (data[0].length !== 3 || data[1].length !== 8)
+          return cb('incorrect buttwoo format, addButtwoo() failed')
+
+        const msgKeyBFE = butt2.hash(data)
+        const msgBIPF = butt2.butt2ToBipf(data, msgKeyBFE)
+
+        const author = bfe.decode(data[1][0])
+        const parent = bfe.decode(data[1][1])
+        const authorParent = author + (parent === null ? '' : parent)
```

Comment:
this is for subfeeds. the top feed would have parent nil. So that works like any other feed, but for subfeeds this is used to distinguish them as they would use the same sequences as another feed for this feed id. it might be good to probably only do this for buttwoo messages

Reply:
This is confusing, can you help me understand? I didn't find anything in the tests in `ssb-buttwoo` for this, and neither in this `ssb-db2` PR.

Suppose a "metafeed" in buttwoo has feed Id `ssb:feed/buttwoo-v1/TBeL1=`. Does this mean that a subfeed of that one has `ssb:feed/buttwoo-v1/TBeL1=` also as the same feedId? Also, what is even "parent"? Is it the feedId of the parent metafeed, or is a message Id? (For what message??)

And why is `authorParent` here only used internally in the codebase? Why couldn't the feedId for this subfeed be the SSB URI `ssb:feed/buttwoo/<FEEDID>/<PARENT>`?

I'll keep asking until I understand it properly, because I think it's important more than one person understands this.

For the `addFeedFormat()` refactor I really need to understand this, because buttwoo works differently than classic and bendy butt and the mechanism I come up with has to work well for buttwoo too.

Reply:
First of all, it is great you are asking these kind of questions. This module and ebt is where we really see how the format plays out :)

I updated the spec, especially around subfeeds to make this more clear (I hope) also what parent is. The big distinction is that feedId and author are potentially different and in classic they are always the same. If we used `ssb:feed/buttwoo/<FEEDID>/<PARENT>` then for the implementation it would be a lot easier, because we would just have 1 field, but then one of the appeals of this, compared to bendy butt would go away: that for example about message would have the same author as the normal messages, so they are easy to look up. That said, some kind of separator between the author and parent would probably be good.

As for tests, the great good thing about the current scheme is that buttwoo doesn't have to care very much about this, you just need to make sure that you supply the correct previous (compared to the feed, not author) and this is really the job of db2 in our stack. So a test case here would be good, I did test it in ebt where we also need to handle this author vs feed in a special way.

All this said, it's clear that subfeeds does complicate things. Besides the changes in ebt & db2, the replication scheduler also needs to know about these. I do really like the idea, but it comes back to how we roll things out. For private groups we'll use and properly test metafeeds. These could be using buttwoo, but would not use the subfeeds feature. Then there is if/when we change the main feed. Here I could see it might make sense, but then again, we already have all the functionality in place to look up things like about messages given a bendy butt subfeed. What I'm saying is, I'm not against removing subfeeds from the spec to keep things simpler. If we do that I'll probably keep the design around under a different name. In any case, I'm curious to hear what you think :)

Reply:
Okay, we're making progress (in my understanding, haha)! Thanks for this. I still don't understand some things, but I can guess what they are doing. I'm going to make a PR to the buttwoo spec and you can let me know if my understanding there is correct or not.